virtcontainers/sandbox.go

```diff
@@ -1105,7 +1105,32 @@ func (s *Sandbox) AddInterface(inf *grpc.Interface) (*grpc.Interface, error) {
 		return nil, err
 	}
 
-	endpoint, err := createVirtualNetworkEndpoint(len(s.networkNS.Endpoints), inf.Name, s.config.NetworkConfig.InterworkingModel)
+	var endpoint Endpoint
```

Comment:
This chunk of code is copied from part of `createEndpointsFromScan`. Please add a new function for it in network.go and just call the new function here.

Reply:
Done. There is a closure.